### DISCUSSION TITLE
Proposal: Change the base package of projects

### DISCUSSION THREAD

**Discussion by soywiz** [korlibs/korge#1249] [Ideas and discussions]
From `com.soywiz....` to just `korge` or whatever.

Proposed here:
https://github.com/korlibs/korge/pull/1185

@ygdrasil-io

**Comment by soywiz** [comment]
PROs:
* Less folders
* Cleaner

CONs:
* Retrocompatibility (since we are in 4.0.0 alphas we can still consider it)

**Reply by ygdrasil-io** [reply]
"There are only two hard things in Computer Science: cache invalidation and naming things." 

Joke aside, auto generated type alias with the old package could, may be, do the trick to offer retrocompatibility.

**Reply by soywiz** [reply]
Do you know how to generate those type aliases automatically?

**Reply by ygdrasil-io** [reply]
I think KSP can do the job, i can try to test it this week end if you want
https://kotlinlang.org/docs/ksp-overview.html

**Reply by ygdrasil-io** [reply]
I created a proof of concept here : 
https://github.com/ygdrasil-io/ksp-alias-generator
With a custom KSP processor : 
https://github.com/ygdrasil-io/ksp-alias-generator/blob/master/processor/src/main/kotlin/alias/generator/AliasSymbolProcessor.kt

On the library, the processor receive two packages from the gradle configuration
```
ksp {
    arg("sourcePackage", "shiny.namespace")
    arg("targetPackage", "old.namespace")
}
```

KSP run the custom processor to generate the code when the library is build and we can call it from the final application

```
fun main() {

    println(old.namespace.decl2
            + old.namespace.sub.namespace.decl2
            + old.namespace.Decl2(3))

}
```

It seem to work on a trivial case, i will try on korge source code later.
